model_metadata.py

```diff
@@ -0,0 +1,61 @@
+import os
+import re
+import json
+import yaml
+from dotenv import load_dotenv
+from azure.ai.ml import MLClient, Input
+from azure.ai.ml.entities import Model
+from azure.ai.ml.constants import AssetTypes
+from azure.identity import DefaultAzureCredential
+
+load_dotenv()
+
+NACHET_SUBSCRIPTION_ID = os.getenv("NACHET_SUBSCRIPTION_ID")
+NACHET_RESOURCE_GROUP = os.getenv("NACHET_RESOURCE_GROUP")
+NACHET_WORKSPACE = os.getenv("NACHET_WORKSPACE")
+
+def generate_model_metadata():
+    """
+    Retrieves deployed online_endpoints and generates metadata json 
+    """
+
+    model_metadata = []
+    model_json = {}
+
+    ml_client = MLClient(DefaultAzureCredential(), NACHET_SUBSCRIPTION_ID, NACHET_RESOURCE_GROUP, NACHET_WORKSPACE)
+
+    # Retrieve all endpoints containing "nachet"
+    endpoints = ml_client.online_endpoints.list()
+    nachet_endpoints = [endpoint for endpoint in endpoints if 'nachet' in endpoint.name.lower()]
+
+    ep = nachet_endpoints[0]
+
+    # Retrieve online_deployment
+    deployment = ml_client.online_deployments.get(endpoint_name=ep.name, name=list(ep.traffic.keys())[0])
+
+    # Retrieve deployment's model (from filePath)
+    model_filepath = deployment.model
+    pattern = re.compile(r"models/([^/]+)/versions/(\d+)")
+    match = pattern.search(model_filepath)
+    if match:
+        model_name = match.group(1)  
+        model_version = match.group(2)  
+    else:
+        print("No match found")
```

Comment:
raise an exception instead (and if you want to print, catch the exception in the caller and print there)

Reply:
Done